posthog/api/plugin.py

```diff
@@ -358,10 +378,31 @@ def update_source(self, request: request.Request, **kwargs):
         for source in PluginSourceFile.objects.filter(plugin=plugin):
             sources[source.filename] = source
         for key, value in request.data.items():
+            transpiled = None
+            error = None
+            status = None
+            try:
+                if key == "site.ts":
+                    transpiled = transpile(value, type="site")
+                    status = PluginSourceFile.Status.TRANSPILED
+                elif key == "frontend.tsx":
+                    transpiled = transpile(value, type="frontend")
+                    status = PluginSourceFile.Status.TRANSPILED
+            except Exception as e:
+                error = str(e)
+                status = PluginSourceFile.Status.ERROR
+
             if key not in sources:
                 performed_changes = True
                 sources[key], created = PluginSourceFile.objects.update_or_create(
-                    plugin=plugin, filename=key, defaults={"source": value}
+                    plugin=plugin,
+                    filename=key,
+                    defaults={
+                        "source": value,
+                        "transpiled": transpiled,
+                        "status": status,
+                        "error": error,
+                    },
                 )
             elif sources[key].source != value:
```

Comment:
don't we need to check if status/transpiled/error got updated too, not only value?

Reply:
Makes sense, thanks